### PR TITLE
Increase test coverage for PQC and CMake

### PR DESCRIPTION
**Pull request by Frauschi**
Increase test coverage, mainly for stuff related to #9605 and #9608 

1. Add more PQC configurations to the CI tests

These are mainly `SMALL` configurations or some edge cases, which have been missing up to now but already caused some build failures (e.g. #9608).

2. More CMake setups tested

Added the functionality added in #9605 to the existing CMake CI tests. Also added a new test to verify the `user_settings.h` functionality with CMake. For that, some missing features have been added to CMake to make the example `example/configs/user_settings_all.h` config file work for the CI test.

### CODE REVIEW

**Comment by Frauschi** [comment]
retest this please

**Comment by Frauschi** [comment]
I think the remaining failing CI tests are unrelated to this PR.

**Comment by dgarske** [comment]
Jenkins retest this please

**Comment by Frauschi** [comment]
Jenkins retest this please

File: cmake/functions.cmake [hunk 1]
```diff
@@ -942,21 +963,17 @@ function(generate_lib_src_list LIB_SOURCES)
         endif()
 
         if(BUILD_FEMATH)
-            if(BUILD_CURVE25519_SMALL)
```
Comment (douzzer) [inline cmake/functions.cmake:945]: Hmmmm, why are you removing this gate?  On non-small builds, this will result in empty translation units, which are warned on pedantic builds, and errors on `-pedantic -Werror` builds.

File: cmake/functions.cmake [hunk 2]
```diff
@@ -942,21 +963,17 @@ function(generate_lib_src_list LIB_SOURCES)
...
                 if(BUILD_INTELASM)
                     list(APPEND LIB_SOURCES wolfcrypt/src/fe_x25519_asm.S)
                 else()
                     list(APPEND LIB_SOURCES wolfcrypt/src/fe_operations.c)
                 endif()
-            endif()
         endif()
 
         if(BUILD_GEMATH)
-            if(BUILD_ED25519_SMALL)
```
Comment (douzzer) [inline cmake/functions.cmake:957]: Same question as above.

File: .github/workflows/pq-all.yml [hunk 3]
```diff
@@ -19,9 +19,14 @@ jobs:
...
+          '--enable-intelasm --enable-sp-asm --enable-all --enable-testcert --enable-acert --enable-dtls13 --enable-dtls-mtu --enable-dtls-frag-ch --enable-dtlscid --enable-quic --with-sys-crypto-policy --enable-experimental --enable-mlkem=yes,kyber,ml-kem --enable-lms --enable-xmss --enable-dilithium --enable-dual-alg-certs --disable-qt CPPFLAGS="-Wdeclaration-after-statement -DWOLFCRYPT_TEST_LINT -DNO_WOLFSSL_CIPHER_SUITE_TEST -DTEST_LIBWOLFSSL_SOURCES_INCLUSION_SEQUENCE" CC=c++',
+          '--enable-intelasm --enable-sp-asm --enable-all --enable-testcert --enable-acert --enable-dtls13 --enable-dtls-mtu --enable-dtls-frag-ch --enable-dtlscid --enable-quic --with-sys-crypto-policy --enable-experimental --enable-mlkem=yes,kyber,ml-kem --enable-lms --enable-xmss --enable-dilithium --disable-qt CPPFLAGS="-pedantic -Wdeclaration-after-statement -DWOLFCRYPT_TEST_LINT -DNO_WOLFSSL_CIPHER_SUITE_TEST -DTEST_LIBWOLFSSL_SOURCES_INCLUSION_SEQUENCE -DWOLFSSL_BLIND_PRIVATE_KEY"',
```
Comment (douzzer) [inline .github/workflows/pq-all.yml:25]: Does it really add meaningful coverage to test a scenario identical to the earlier one, but without `--enable-dual-alg-certs`?

File: .github/workflows/pq-all.yml [hunk 4]
```diff
@@ -19,9 +19,14 @@ jobs:
...
+          '--enable-intelasm --enable-sp-asm --enable-all --enable-testcert --enable-acert --enable-dtls13 --enable-dtls-mtu --enable-dtls-frag-ch --enable-dtlscid --enable-quic --with-sys-crypto-policy --enable-experimental --enable-mlkem=yes,kyber,ml-kem --enable-lms --enable-xmss --enable-dilithium --enable-dual-alg-certs --disable-qt CPPFLAGS="-pedantic -Wdeclaration-after-statement -DWOLFCRYPT_TEST_LINT -DNO_WOLFSSL_CIPHER_SUITE_TEST -DTEST_LIBWOLFSSL_SOURCES_INCLUSION_SEQUENCE -DWOLFSSL_BLIND_PRIVATE_KEY"',
```
Comment (douzzer) [inline .github/workflows/pq-all.yml:26]: Should `-DWOLFSSL_BLIND_PRIVATE_KEY` just be added to the existing scenarios above?  Or do you think it's meaningful to test with and without it?

File: .github/workflows/pq-all.yml [hunk 5]
```diff
@@ -19,9 +19,14 @@ jobs:
...
+          '--enable-intelasm --enable-sp-asm --enable-all --enable-testcert --enable-acert --enable-dtls13 --enable-dtls-mtu --enable-dtls-frag-ch --enable-dtlscid --enable-quic --with-sys-crypto-policy --enable-experimental --enable-mlkem=yes,kyber,ml-kem --enable-lms --enable-xmss --enable-dilithium --enable-dual-alg-certs --disable-qt CPPFLAGS="-pedantic -Wdeclaration-after-statement -DWOLFCRYPT_TEST_LINT -DNO_WOLFSSL_CIPHER_SUITE_TEST -DTEST_LIBWOLFSSL_SOURCES_INCLUSION_SEQUENCE -DWOLFSSL_TLSX_PQC_MLKEM_STORE_OBJ"',
```
Comment (douzzer) [inline .github/workflows/pq-all.yml:27]: Ditto for `-DWOLFSSL_TLSX_PQC_MLKEM_STORE_OBJ`.

**Review by douzzer — CHANGES_REQUESTED** [review]
Minor questions+quibbles.

Looks good otherwise.

File: .github/workflows/pq-all.yml (hunk 3 above)
Comment (Frauschi) [reply .github/workflows/pq-all.yml:25]: Yes, I think so. Both the `dual-alg-certs` option as well as the `WOLFSSL_BLIND_PRIVATE_KEY` option enable new code paths but also disable others. Hence, enabling and disabling both individually compared to the base case results in different code paths, which are not covered otherwise in the tests.

File: .github/workflows/pq-all.yml (hunk 4 above)
Comment (Frauschi) [reply .github/workflows/pq-all.yml:26]: See comment above for `dual-alg-certs`.

File: .github/workflows/pq-all.yml (hunk 5 above)
Comment (Frauschi) [reply .github/workflows/pq-all.yml:27]: This option could also be added to one of the new cases above. However, in this scenario, when the test then fails, one has to check whether the fail results from the `WOLFSSL_TLSX_PQC_MLKEM_STORE_OBJ` option or from the other one to where it would be added.

File: cmake/functions.cmake (hunk 1 above)
Comment (Frauschi) [reply cmake/functions.cmake:945]: The motivation to remove the gate was that when using the user_settings.h approach with CMake, both the `fe_low_mem.c` and the `fe_operations.c` files are required to succeed the build. In this case, pedantic builds are impossible anyway as the user_settings.h approach uses empty translation units throughout.

I re-added the gates but added logic to also build the small source files in case `WOLFSSL_USER_SETTINGS` is set. Hence, empty translation units are only caused when the user_settings.h approach is used.

File: cmake/functions.cmake (hunk 2 above)
Comment (Frauschi) [reply cmake/functions.cmake:957]: See above

**Comment by Frauschi** [comment]
> Minor questions+quibbles.

I addressed the CMake gate thing in the updated commit. The options in the PQ build are beneficial in my view, see comments.

**Comment by Frauschi** [comment]
Retest this please

**Comment by Frauschi** [comment]
Jenkins restest this please
